routing/routing_session.cpp

```diff
@@ -786,6 +755,104 @@ void RoutingSession::CopyTraffic(traffic::AllMwmTrafficInfo & trafficColoring) c
   TrafficCache::CopyTraffic(trafficColoring);
 }
 
+void RoutingSession::ProcessSpeedCameras(GpsInfo const & info)
+{
+  CHECK_THREAD_CHECKER(m_threadChecker, ());
+
+  auto const passedDistanceMeters = m_route->GetCurrentDistanceFromBeginMeters();
+
+  // Step 1. Find new cameras, to be cached.
+  size_t firstNotChecked = m_lastNotCheckedSpeedCameraIndex;
+  if (firstNotChecked == m_route->GetRouteSegments().size())
+    return;
+
+  CHECK_LESS(firstNotChecked, m_route->GetRouteSegments().size(), ());
+
+  auto const & segments = m_route->GetRouteSegments();
+  double distToPrevSegment = segments[firstNotChecked].GetDistFromBeginningMeters();
+  double distFromCurPosToLatestCheckedSegmentM = distToPrevSegment - passedDistanceMeters;
+
+  while (firstNotChecked < segments.size() &&
+         distFromCurPosToLatestCheckedSegmentM < SpeedCameraOnRoute::kLookAheadDistanceMeters)
+  {
+    auto const & lastSegment = segments[firstNotChecked];
+    auto const & speedCamsVector = lastSegment.GetSpeedCams();
+    double segmentLength = m_route->GetSegLenMeters(firstNotChecked);
+    if (!speedCamsVector.empty())
+    {
+      size_t n = speedCamsVector.size();
+      for (size_t curCameraIndex = 0; curCameraIndex < n; ++curCameraIndex)
```

Comment:
range-based for
also, you can now drop the empty() check and the variable for n

Reply:
done